grafana/resource_alerting_mute_timing.go

```diff
@@ -0,0 +1,307 @@
+package grafana
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceMuteTiming() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+* [Official documentation](https://grafana.com/docs/grafana/next/alerting/notifications/mute-timings/)
+* [HTTP API](https://grafana.com/docs/grafana/next/developers/http_api/alerting_provisioning/#mute-timings)
+		`,
+
+		CreateContext: createMuteTiming,
+		ReadContext:   readMuteTiming,
+		UpdateContext: updateMuteTiming,
+		DeleteContext: deleteMuteTiming,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the mute timing.",
+			},
+
+			"intervals": {
+				Type:        schema.TypeList,
```

Comment:
A list rather than a set here is fairly nuanced. We need to reply on diff-suppression deeper in this file, but turns out that diff-suppression is completely broken in Terraform if there is a set anywhere on the path to that field.

This happens because TF represents sets inside the state object using the set's hash as a key, like:
```
intervals: map[string]interface{} {
    "hash-of-elem-1": <elem-1-data>,
    "hash-of-elem-2": <elem-2-data>,
}
```

Since the hash is the key, a change on any field no matter how deeply nested will cause the hash to be different. Terraform will think that this is a deletion and creation of the member at every layer, not an update.

This breaks the diffing (which we rely on for month->month-number conversions). This core reason is also coincidentally why the output of `tf plan` is so useless on sets, it shows every element as being changed rather than the one that actually got changed. Interesting issue! We just use lists to work around it.

Reply:
Nit: should we add this as a comment above the field?